Geometry_Engine/Query/DiscontinuityPoints.cs

```diff
@@ -34,31 +34,57 @@ public static partial class Query
         /**** Public Methods - Curves                   ****/
         /***************************************************/
 
-        public static List<Point> DiscontinuityPoints(this Arc curve)
+        [PreviousVersion("3.3", "BH.Engine.Geometry.Query.DiscontinuityPoints(BH.oM.Geometry.Arc)")]
+        public static List<Point> DiscontinuityPoints(this Arc curve, double distanceTolerance = Tolerance.Distance, double angleTolerance = Tolerance.Angle)
         {
             return new List<Point> { curve.StartPoint(), curve.EndPoint() };
```

Comment:
Wouldn't it be worth checking if the arc's angle is not a multiplication of 2*Pi? Happy to resolve it in a separate thread as out of the scope of this PR.

Reply:
Hm yes, if we assume that any 2*Pi Arc is a Circle it shouldn't return any discontinuity points. But do we?

Also I've taken a look at the `Arc` to `Circle` casting operation:      
```
public static explicit operator Circle(Arc curve)
{
    if (System.Math.Abs((curve.EndAngle - curve.StartAngle) - System.Math.PI * 2) > Tolerance.Distance)
        return null;

        return new Circle()
    {
        Centre = curve.CoordinateSystem.Origin,
        Normal = curve.CoordinateSystem.Z,
        Radius = curve.Radius,
    };
}
```
Shouldn't it record any `couldn't convert arc to circle` error?
If we try to cast a regular arc to a circle it can mess results without letting the user know that something went wrong.
![image](https://user-images.githubusercontent.com/48125514/92127773-ef932f80-ee01-11ea-88a9-5a6a6203d0ec.png)

Reply:
> Hm yes, if we assume that any 2*Pi Arc is a Circle it shouldn't return any discontinuity points. But do we?

In my understanding, if tangents of two meeting segments (or one, in case of 360 deg arc) are same, then there is no discontinuity at that point. However, it all comes down to how we define the discontinuity.

Concerning the `Arc` => `Circle` cast, shouldn't the component turn red due to `null` argument?

Reply:
> shouldn't the component turn red due to null argument? 

It doesn't. The method always just returns an empty list so the null argument is never used (and stays undiscovered).
![image](https://user-images.githubusercontent.com/48125514/92227485-a3e69180-eea6-11ea-9ead-1cb1892e58ed.png)

Reply:
Right, good point. Then there should be a null check I guess

Reply:
Moving to #1964